src/tracker_service.py

```diff
@@ -0,0 +1,254 @@
+from viam.proto.common import PointCloudObject, ResourceName
+from viam.proto.service.vision import Classification, Detection
+from viam.resource.base import ResourceBase
+from viam.resource.types import Model, ModelFamily
+from viam.services.mlmodel import MLModel, MLModelClient
+from viam.services.vision import CaptureAllResult, Vision, VisionClient
+from viam.utils import ValueTypes
+
+from src.config.config import TrackerConfig
+from src.test.fake_embedder_ml_model_service import FakeEmbedderMLModel
+from src.tracker.detector.custom_vision_service_detector import (
+    CustomVisionServiceDetector,
+)
+from src.tracker.detector.detector import Detector
+from src.tracker.detector.torchvision_detector import TorchvisionDetector
+from src.tracker.embedder.custom_mlmodel_service_embedder import (
+    CustomMLModelServiceEmbedder,
+)
+from src.tracker.embedder.embedder import Embedder
+from src.tracker.tracker import Tracker
+
+LOGGER = getLogger(__name__)
+
+
+class TrackerService(Vision, Reconfigurable):
+    """TrackerService is a subclass a Viam Vision Service"""
+
+    MODEL: ClassVar[Model] = Model(
+        ModelFamily("viam", "vision"), "camera-object-tracking-service"
+    )
+
+    def __init__(self, name: str):
+        super().__init__(name=name)
+        self.camera: CameraClient = None
+        self.detector: Detector = None
+        self.embedder: Embedder = None
+        self.tracker = None
+
+    @classmethod
+    def new_service(
+        cls, config: ServiceConfig, dependencies: Mapping[ResourceName, ResourceBase]
+    ) -> Self:
+        """returns new vision service"""
+        service = cls(config.name)
+        service.reconfigure(config, dependencies)
+        return service
+
+    # Validates JSON Configuration
+    @classmethod
+    def validate_config(cls, config: ServiceConfig) -> Sequence[str]:
+        """Validate config and returns a list of dependencies."""
+        dependencies = []
+        camera_name = config.attributes.fields["camera_name"].string_value
+        dependencies.append(camera_name)
+        detector_name = config.attributes.fields["detector_name"].string_value
+        if detector_name:
+            dependencies.append(detector_name)
+        embedder_name = config.attributes.fields["embedder_name"].string_value
+        if embedder_name:
+            dependencies.append(embedder_name)
+
+        # validate the config
+        _ = TrackerConfig(config)
+        return dependencies
+
+    def reconfigure(
+        self, config: ServiceConfig, dependencies: Mapping[ResourceName, ResourceBase]
+    ):
+        tracker_cfg = TrackerConfig(config)
+        self.camera_name = config.attributes.fields["camera_name"].string_value
+        self.camera = dependencies[Camera.get_resource_name(self.camera_name)]
+        detector_name = config.attributes.fields["detector_name"].string_value
+
+        if not detector_name:
+            self.detector = TorchvisionDetector(tracker_cfg.detector_config)
```

Comment:
Kinda like the logger warning when starting a default.  Can you do it here for the detector too?

Reply:
you need to add "nit" before such comments

Reply:
done